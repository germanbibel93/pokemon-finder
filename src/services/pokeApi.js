/* eslint-disable import/prefer-default-export */
import axios from 'axios';

export const PokeApi = async () =>{ 
    const resJson = await axios
    .get(`https://pokeapi.co/api/v2/pokemon?limit=807`)
    
    .then(function (response) {
        console.log("get avanzo")
      const { data } = response;
      const { results } = data;
      const newPokemonData = {};
      results.forEach((pokemon, index) => {
        newPokemonData[index + 1] = {
          id: index + 1,
          name: pokemon.name,
          sprite: `https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/${
            index + 1
          }.png`,
        };
      });
      return newPokemonData;
    });
    return resJson
}
