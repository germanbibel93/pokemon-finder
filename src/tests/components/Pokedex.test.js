import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
 
import Pokedex from '../../components/Layout';
import { PokeApi } from '../../services/pokeApi';
import PokeList from '../mock/pokeList';

jest.mock('../../services/pokeApi');


describe('Pokedex-test', () => {
  test('renders Pokedex component', () => {
    PokeApi.mockReturnValueOnce(Promise.resolve(PokeList));
    render(<Pokedex />);
    expect(screen.getByText('Buscar')).toBeInTheDocument();
    
  });

  test('Pokemon search', async () => {
    PokeApi.mockReturnValueOnce(Promise.resolve(PokeList));
    const {findByText } = render(<Pokedex />)
    expect(screen.queryByTestId('pokedex-grid')).not.toBeInTheDocument();

    expect(screen.getByTestId('search-textarea')).toBeInTheDocument();
    const search = screen.getByTestId('search-textarea');
    fireEvent.change(search, { target: { value: 'man' } });
    const grid = { classes: { root: 'root' } };
    expect(grid).toMatchSnapshot();
    const items = await findByText('4. Charmander')
    expect(items).toBeInTheDocument();
  });

});