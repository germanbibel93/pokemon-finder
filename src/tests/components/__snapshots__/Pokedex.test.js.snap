// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App M 1`] = `
Object {
  "classes": Object {
    "root": "root",
  },
}
`;

exports[`Pokedex-test Pokemon search 1`] = `
Object {
  "classes": Object {
    "root": "root",
  },
}
`;
